scenarioo-client/app/pipes/metadata/treeDataOptimizer.pipe.spec.ts

```diff
@@ -0,0 +1,320 @@
+/* scenarioo-client
+ * Copyright (C) 2014, scenarioo.org Development Team
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {TreeDataOptimizerPipe} from './treeDataOptimizer.pipe';
+
+describe('Pipe: scTreeDataOptimizer', () => {
+    let scTreeDataOptimizer: TreeDataOptimizerPipe;
+    let output;
+
+    const DATA_EMPTY_NODE = {
+        childNodes: [
+            {
+                nodeLabel: 'keyTwo',
+                nodeValue: 'valueTwo',
+            },
+            {
+                nodeLabel: 'empty',
+                childNodes: [],
+            },
+        ],
+    };
+
+    const DATA_EMPTY_NODE_OPTIMIZED = {
+        childNodes: [
+            {
+                nodeLabel: 'keyTwo',
+                nodeValue: 'valueTwo',
+            },
+        ],
+    };
+
+    const DATA_DETAILS = {
+        childNodes: [
+            {
+                nodeLabel: 'details',
+                childNodes: [
+                    {
+                        nodeLabel: 'hero',
+                        nodeValue: 'Donald Duck',
+                    },
+                ],
+            },
+        ],
+    };
```

Comment:
Just a proposal to improve readability of the tests (I know this is not your fault -as you did not write this code in a first place): Since all of this constants are just used in one test I would rather prefer to define the consts directly in the place where they are used (thus inside the tests), as it keeps the tests more readable directly in one place without having to jump forth an back.